humanoid_league_team_communication/humanoid_league_team_communication/converter/robocup_protocol_converter.py

```diff
@@ -0,0 +1,74 @@
+from enum import IntEnum
+
+from humanoid_league_team_communication.converter.message_to_team_data_converter import MessageToTeamDataConverter
+from humanoid_league_team_communication.converter.state_to_message_converter import StateToMessageConverter
+
+from humanoid_league_msgs.msg import ObstacleRelative, Strategy
+from soccer_vision_attribute_msgs.msg import Robot as RobotAttributes
+
+import humanoid_league_team_communication.robocup_extension_pb2 as Proto
+
+
+class TeamColor(IntEnum):
+    BLUE = 0
+    RED = 1
+
+
+class RobocupProtocolConverter:
+
+    def __init__(self, own_team_color: TeamColor):
+        self.role_mapping = (
+            (Proto.Role.ROLE_UNDEFINED, Strategy.ROLE_UNDEFINED),
+            (Proto.Role.ROLE_IDLING, Strategy.ROLE_IDLING),
+            (Proto.Role.ROLE_OTHER, Strategy.ROLE_OTHER),
+            (Proto.Role.ROLE_STRIKER, Strategy.ROLE_STRIKER),
+            (Proto.Role.ROLE_SUPPORTER, Strategy.ROLE_SUPPORTER),
+            (Proto.Role.ROLE_DEFENDER, Strategy.ROLE_DEFENDER),
+            (Proto.Role.ROLE_GOALIE, Strategy.ROLE_GOALIE),
+        )
+        self.action_mapping = (
+            (Proto.Action.ACTION_UNDEFINED, Strategy.ACTION_UNDEFINED),
+            (Proto.Action.ACTION_POSITIONING, Strategy.ACTION_POSITIONING),
+            (Proto.Action.ACTION_GOING_TO_BALL, Strategy.ACTION_GOING_TO_BALL),
+            (Proto.Action.ACTION_TRYING_TO_SCORE, Strategy.ACTION_TRYING_TO_SCORE),
+            (Proto.Action.ACTION_WAITING, Strategy.ACTION_WAITING),
+            (Proto.Action.ACTION_KICKING, Strategy.ACTION_KICKING),
+            (Proto.Action.ACTION_SEARCHING, Strategy.ACTION_SEARCHING),
+            (Proto.Action.ACTION_LOCALIZING, Strategy.ACTION_LOCALIZING),
+        )
+        self.side_mapping = (
+            (Proto.OffensiveSide.SIDE_UNDEFINED, Strategy.SIDE_UNDEFINED),
+            (Proto.OffensiveSide.SIDE_LEFT, Strategy.SIDE_LEFT),
+            (Proto.OffensiveSide.SIDE_MIDDLE, Strategy.SIDE_MIDDLE),
+            (Proto.OffensiveSide.SIDE_RIGHT, Strategy.SIDE_RIGHT),
+        )
+
+        self.proto_to_team_data_team_mapping = {
+            Proto.Team.UNKNOWN_TEAM: ObstacleRelative.ROBOT_UNDEFINED,
+            Proto.Team.BLUE: ObstacleRelative.ROBOT_CYAN,
+            Proto.Team.RED: ObstacleRelative.ROBOT_MAGENTA
+        }
+        # @TODO: check if RobotAttributes.TEAM_OWN is based on the robot seen or robot running
```

Comment:
Well I thought since we are talking about a "seen" robot as we are using vision messages, it could be that the `TEAM_OWN` of a "seen" red robot is the ID for the red team  as the message represents that robot, even if we are running this code on our robot in the blue team.
So I just wanted to make sure, how those attributes are initialized

Reply:
Okay. Yes, the vision queries our team color and labels the red / blue robot accordingly as the own / opponent team. This is a change that was done when we migrated to the soccer_vision_msgs.